### PR TITLE
unixdatetime: add new type

### PR DESCRIPTION
**Pull request by Viterkim**

### CODE REVIEW

File: src/validators/unixdatetime.test.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+        {
+          value: '',
+          expectedErrors: [new WrongLengthFail('Must contain between 1 and 20 characters', '')]
+        },
+        {
+          value: 'abcdef',
+          expectedErrors: [new NotIntegerFail('Must be a number in string format', 'abcdef')]
+        },
+        {
+          value: '2147483648',
```
Comment (ofweb) [inline src/validators/unixdatetime.test.ts:32]: are there any reason to limit to 32bit?

File: src/validators/unixdatetime.test.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+          expectedErrors: [new OutOfRangeFail('Must be a valid Unix timestamp', '2147483648')]
+        },
+        {
+          value: '-1',
+          expectedErrors: [new OutOfRangeFail('Must be a valid Unix timestamp', '-1')]
+        },
+        {
+          value: '123.45',
+          expectedErrors: [new NotIntegerFail('Must be a number in string format', '123.45')]
```
Comment (ofweb) [inline src/validators/unixdatetime.test.ts:41]: The error message is a bit wierd here.

File: src/validators/unixdatetime.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+export abstract class UnixDateTimeValidator<O = never> extends ValidatorBase<string | O> {
+  public constructor(options?: ValidatorBaseOptions) {
+    super(options)
+    if (options?.optimize !== false) {
+      this.optimize()
+    }
+  }
+
+  public codeGen(
```
Comment (ofweb) [inline src/validators/unixdatetime.ts:55]: I could not find any tests for this.
Normally the tests test both the optimized code and the not optimized code.

File: src/validators/unixdatetime.ts [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+export function validateUnixDateTime(value: unknown, context?: string): ValidationFailure[] {
+  const stringError = validateString(value, 1, 20, context)
+  if (!isValidType<string>(value, stringError)) {
+    return stringError
+  }
+
+  const numValue = Number(value)
+
+  if (isNaN(numValue) || !Number.isInteger(numValue)) {
```
Comment (ofweb) [inline src/validators/unixdatetime.ts]: we also need to check for `isFinite` here

File: src/validators/unixdatetime.ts [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+  return false
+}
+
+export function validateUnixDateTime(value: unknown, context?: string): ValidationFailure[] {
+  const stringError = validateString(value, 1, 20, context)
+  if (!isValidType<string>(value, stringError)) {
+    return stringError
+  }
+
+  const numValue = Number(value)
```
Comment (ofweb) [inline src/validators/unixdatetime.ts:33]: That is not a good way to convert a string to a number. please read the docs here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
I would suggest using this instead: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt

**Review by ofweb — CHANGES_REQUESTED** [review]
It might be time to read this wiki page again: https://wiki.connectedcars.io/en/tech/productivity-tools

File: src/validators/unixdatetime.test.ts (hunk 1 above)
Comment (Viterkim) [reply src/validators/unixdatetime.test.ts:32]: unix timestamps are 32 bit

**Review by ofweb — APPROVED** [review]

**Pull request merged**